### PR TITLE
nxp: imx8ulp: switch to Zephyr native drivers and timer domain

### PR DESCRIPTION
**Pull request by LaurentiuM1234**

### CODE REVIEW

File: app/boards/imx8ulp_evk_mimx8ud7_adsp.overlay [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	rx-sync-mode = <1>;
+	status = "okay";
+};
+
+/*
+ * TODO: this node shouldn't be here. For now, we only support
+ * the EVK board in Zephyr, while in SOF we support the EVK and
+ * the EVK9 boards. This is an issue because the two boards use
+ * different SAI instances for BT communication (SAI5 - EVK,
+ * SAI6 - EVK). Ideally, we should have two boards in Zephyr
```
[the remark this answers is not in the document]
Comment (LaurentiuM1234) [reply app/boards/imx8ulp_evk_mimx8ud7_adsp.overlay]: yep, sry, fixed

**Review by iuliana-prodan — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Comment by LaurentiuM1234** [comment]
Testing done. Dropping `DNM` tag.

**Comment by kv2019i** [comment]
First time I see " PR has merge conflicts! Need to rebase to continue test! " in sof-ci jenkins build in a case where github is not reporting merge conflicts. Let me rekick the SOF CI.

**Comment by kv2019i** [comment]
SOFCI TEST

**Pull request merged**